tests/backends/sqlite/tests.py

```diff
@@ -245,3 +246,55 @@ def create_object():
             for conn in thread_connections:
                 if conn is not main_connection:
                     conn.close()
+
+
+@unittest.skipUnless(connection.vendor == "sqlite", "SQLite tests")
+@override_settings(DEBUG=True)
+class TestTransactionMode(TransactionTestCase):
+    available_apps = ["backends"]
+
+    def test_default_transaction_mode(self):
+        with transaction.atomic():
+            pass
+        self.assertEqual(connection.queries[0]["sql"], "BEGIN")
+
+    def test_invalid_transaction_mode(self):
+        connection = self._get_connection_for_transaction_mode("invalid")
+        with self.assertRaises(ImproperlyConfigured):
+            connection.ensure_connection()
```

Comment:
Use `override_settings` here instead and assert against the message as well using `assertRaisesMessage`

Reply:
I did try to use override_settings initially but it didn't work. I spend some more time on it trying to figure out why, but I think it's because transaction test case itself uses the DATABASES setting to bootstrap itself up.

Reply:
That's the case but only for the `DATABASE['default']` entry by default. Try doing something like

```python
with self.modify_settings(DATABASES=settings.DATABASES | settings_dict)
``` 

by making sure to use a different alias than `"default"` and returning `connections[test_alias]` instead.

That might force you to make `_get_connection_for_transaction_mode` a context manager but that could be used to close the connection immediately after the test as well.

If that doesn't work it's not the end of the world, your approach is relatively clean, but we usually try to avoid using internal APIs if possible during tests as internal restructuration then requires adjusting them and it could result in a loss of intent from the original test.

Reply:
Seems like postgres backend tests [use a different approach where they replace `settings_dict`](https://github.com/django/django/blob/4879907223d70ee1a82474d9286ccfa5dae96971/tests/backends/postgresql/test_creation.py#L22-L38) and rely on `SimpleTestCase` to avoid the pre-existing connection creation.

Reply:
Similar situation with [the mysql tests](https://github.com/django/django/blob/4879907223d70ee1a82474d9286ccfa5dae96971/tests/backends/mysql/test_creation.py#L61-L73).

So maybe the right approach here is to use `SimpleTestCase` with `databases = {"default"}` to prevent implicit connection creation, make sure to close it on `tearDown`, and get more freedom as your test methods both test failed connection creation and the generated SQL from `atomic`.

Reply:
I have updated the PR based on your suggestions, I think it looks cleaner now, but let me know if we can make it even better.